spectral_cube/stokes_spectral_cube.py

```diff
@@ -34,8 +50,14 @@ def __init__(self, stokes_data, mask=None, meta=None, fill_value=None):
 
         reference = tuple(stokes_data.keys())[0]
 
-        for component in stokes_data:
+        # Validate and map Stokes components using StokesCoord, with custom mapping
+        try:
+            with custom_stokes_symbol_mapping(self._custom_stokes_map):
+                stokes_coord = StokesCoord(list(stokes_data.keys()))
+        except Exception as e:
+            raise ValueError(f"Invalid Stokes components: {e}")
```

Comment:
Why are we catching a generic exception here?  Let's try not to do that if we can avoid it; this could otherwise unintentionally catch, e.g., interrutps